Client/SafeBrowsing/GSafeBrowsing.swift

```diff
@@ -0,0 +1,316 @@
+    
+    private let baseURL = "https://safebrowsing.brave.com"
+    private let session = URLSession(configuration: .ephemeral)
+    private let database = SafeBrowsingDatabase()
+    
+    public static let shared = SafeBrowsingClient()
+    
+    private init() {
+    }
+    
+    func find(_ hashes: [String], _ completion: @escaping (_ isSafe: Bool, Error?) -> Void) {
+        let discoveredHashes = database.find(hashes)
+        if discoveredHashes.isEmpty {
+            return completion(true, nil)
+        }
+        
+        let clientInfo = ClientInfo(clientId: SafeBrowsingClient.clientId,
+                                    clientVersion: SafeBrowsingClient.version)
+        
+        let threatTypes: [ThreatType] = [.malware,
+                                         .socialEngineering,
+                                         .unwantedSoftware,
+                                         .potentiallyHarmfulApplication]
+        
+        let platformTypes: [PlatformType] = [.any]
+        let threatEntryTypes: [ThreatEntryType] = [.url, .exe]
+        
+        let threatInfo = ThreatInfo(threatTypes: threatTypes,
+                                    platformTypes: platformTypes,
+                                    threatEntryTypes: threatEntryTypes,
+                                    threatEntries: discoveredHashes.map {
+                                        return ThreatEntry(hash: $0, url: nil, digest: nil)
+                                    }
+        )
+        
+        do {
+            let body = FindRequest(client: clientInfo, threatInfo: threatInfo)
+            let request = try encode(.post, endpoint: .fullHashes, body: body)
+            executeRequest(request, type: FindResponse.self) { response, error in
+                if let error = error {
+                    return completion(false, error)
+                }
+                
+                if let response = response {
+                    if !response.matches.isEmpty {
+                        return completion(false, nil)
+                    }
+                }
+                
+                completion(true, nil)
+            }
+        } catch {
+            completion(false, error)
+        }
+    }
+    
+    func fetch(_ completion: @escaping (Error?) -> Void) {
+        let clientInfo = ClientInfo(clientId: SafeBrowsingClient.clientId,
+                                    clientVersion: SafeBrowsingClient.version)
+        
+        let constraints = Constraints(maxUpdateEntries: UInt32(SafeBrowsingClient.maxBandwidth),
+                                      maxDatabaseEntries: UInt32(SafeBrowsingClient.maxDiskSpace),
+                                      region: "US",
+                                      supportedCompressions: [.raw],
+                                      language: nil,
+                                      deviceLocation: nil)
+        
+        let lists = [
+            ListUpdateRequest(threatType: .malware,
```

Comment:
`potentiallyHarmfulApplication` is missing here

Reply:
Whenever I add that, the calls fail. I actually copied the same exact requests as Google’s Go-Lang implementation. iOS doesn’t allow you to download malicious application. I believe that’s the reason for the error but I can try it again and see.

Reply:
I think potentially harmful applications is an Android-only list so that would make sense. Whenever you request a list that's not available on your platform, the update fails.